packages/k2-alpine/src/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -163,22 +168,37 @@ const Segment = ({
   onPress
 }: {
   sx?: SxProp
-  ratio?: number
+  ratio: number | DerivedValue<number[]>
   text: string
   index: number
   selectedIndex: SharedValue<number>
   backgroundColor?: string
-  onTextWidthChange: (width: number) => void
+  onTextWidthChange?: (index: number, width: number) => void
   onPress: () => void
 }): JSX.Element => {
   const { theme } = useTheme()
 
-  const handleTextLayout = (event: LayoutChangeEvent): void => {
-    onTextWidthChange(event.nativeEvent.layout.width + 64)
-  }
+  const throttledOnPress = useMemo(
+    () =>
+      throttle(onPress, THROTTLE_DURATION, {
+        leading: true,
+        trailing: false
+      }),
+    [onPress]
+  )
```

Comment:
added some throttle here to make sure we don't keep clicking it, which results in weird animation/state